config/ompi_setup_mpi_fortran.m4

```diff
@@ -376,6 +377,20 @@ end program]])],
           [OMPI_FORTRAN_BUILD_SIZEOF=0])
     AC_SUBST(OMPI_FORTRAN_BUILD_SIZEOF)
 
+    AC_ARG_ENABLE([deprecate-mpif-h],
+                  [AS_HELP_STRING([--enable-deprecate-mpif-h],
+                                  [Mark the mpif.h bindings as deprecated (default: enabled)])])
+    AS_IF([! test x"$enable_deprecate_mpif_h" = "xno"],
+            [OMPI_FORTRAN_CHECK_WARNING([enable_deprecate_mpif_h="yes"],
+                                        [enable_deprecate_mpif_h="no"])])
+    AC_MSG_CHECKING([if we mark mpif.h bindings as deprecated])
+    AS_IF([test "$enable_deprecate_mpif_h" = "yes"],
+            [OMPI_FORTRAN_DEPRECATE_MPIF_H="#warning mpif.h is deprecated since MPI 4.1. Refer to MPI Sec. 19.1.4."
+             AC_MSG_RESULT([yes])],
+            [OMPI_FORTRAN_DEPRECATE_MPIF_H=""
+             AC_MSG_RESULT([no])])
```

Comment:
My $0.02: I'd put this in ompi_fortran_check_warning_deprecated.m4.  I.e., have all the handling of it in one place.  For example, if someday we have to handle multiple different preprocessor directives for warnings across different compilers, that one place can figure out what the string is to substitute into `mpif.h`.